js/app/src/Blocks.svelte

```diff
@@ -417,6 +418,19 @@
 					handled_dependencies[i].push(id);
 				});
 		});
+		if (!attached_error_listeners) {
+			attached_error_listeners = true;
+			components.forEach((c) => {
+				if (c.instance) {
+					c.instance.$on("error", (event_data: any) => {
+						messages = [
+							new_message(event_data.detail, -1, "error"),
+							...messages
+						];
+					});
+				}
+			});
+		}
```

Comment:
Couldn't this potentially only attach a single listener when there could be multiple?

Reply:
So instead of only running this once, every time we should track which components have an error listener attached, and add to the ones that don't, correct?

Reply:
Yes, sorry (missed this earlier).